cmd/blockstorage_update.go

```diff
@@ -0,0 +1,113 @@
+package cmd
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/spf13/cobra"
+
+	"github.com/exoscale/cli/pkg/globalstate"
+	"github.com/exoscale/cli/pkg/output"
+	v3 "github.com/exoscale/egoscale/v3"
+)
+
+type blockstorageUpdateCmd struct {
+	cliCommandSettings `cli-cmd:"-"`
+
+	_ bool `cli-cmd:"update"`
+
+	Name   string            `cli-arg:"#" cli-usage:"NAME|ID"`
+	Size   int64             `cli-usage:"block storage volume size"`
+	Labels map[string]string `cli-flag:"label" cli-usage:"block storage volume label (format: key=value)"`
+	Zone   v3.ZoneName       `cli-short:"z" cli-usage:"block storage zone"`
+}
+
+func (c *blockstorageUpdateCmd) cmdAliases() []string { return gCreateAlias }
+
+func (c *blockstorageUpdateCmd) cmdShort() string { return "Update a Block Storage Volume" }
+
+func (c *blockstorageUpdateCmd) cmdLong() string {
+	return fmt.Sprintf(`This command Updates Block Storage Volume.
+
+Supported output template annotations: %s`,
+		strings.Join(output.TemplateAnnotations(&blockstorageShowOutput{}), ", "))
+}
+
+func (c *blockstorageUpdateCmd) cmdPreRun(cmd *cobra.Command, args []string) error {
+	cmdSetZoneFlagFromDefault(cmd)
+	return cliCommandDefaultPreRun(c, cmd, args)
+}
+
+func (c *blockstorageUpdateCmd) cmdRun(_ *cobra.Command, _ []string) error {
+	ctx := gContext
+	client, err := switchClientZoneV3(ctx, globalstate.EgoscaleV3Client, c.Zone)
+	if err != nil {
+		return err
+	}
+
+	volumes, err := client.ListBlockStorageVolumes(ctx)
+	if err != nil {
+		return err
+	}
+
+	volume, err := volumes.FindBlockStorageVolume(c.Name)
+	if err != nil {
+		return err
+	}
+
+	if len(c.Labels) == 0 && c.Size == 0 {
```

Comment:
What are you guarding against here? shouldn't you return an error?

Reply:
If nothing to update, nothing to output then, I want to avoid displaying the decorator for nothing if no flag provided.
But we can remove this statement, and we will have a short time display of:

```
Updating block storage volume test-volume...
```

No strong opinion here :)